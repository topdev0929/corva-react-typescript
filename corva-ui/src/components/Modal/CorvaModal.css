.small {
  width: 416px;
}

.medium {
  width: 552px;
}

.large {
  width: 768px;
}

.extraLarge {
  width: 80%;
  height: calc(100% - 32px);
  max-width: 1152px;
}

.modalWrapper {
  display: flex;
  width: 100%;
  height: 100%;
}

/* Container */
.modalContainer {
  background-color: var(--palette-background-b-9);
  border-radius: 4px;
  display: flex;
  flex-direction: column;
  margin: auto;
  max-height: calc(100% - 32px);
  max-width: 80%;
  outline: none;
  position: relative;
  overflow: hidden;
}

.modalContainerMobile {
  max-height: calc(100% - 16px);
  max-width: calc(100% - 16px);
  min-height: calc(100% - 16px);
  min-width: calc(100% - 16px);
}

/* Title */
.title {
  align-items: center;
  background-color: var(--palette-background-b-9);
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  box-shadow: 0px 8px 10px var(--palette-background-b-9);
  display: flex;
  justify-content: space-between;
  line-height: 28px;
  padding: 40px 40px 0;
  z-index: 1;
  width: 100%;
}

.titleMobile {
  padding: 12px 16px;
}

.titleText {
  color: white;
  font-size: 20px;
}

/* Modal content */
.modalContent {
  overflow-y: auto;
  padding: 16px 40px;
}

.modalContentMobile {
  background-color: inherit;
  overflow-y: scroll;
  padding: 16px;
}

/* Modal actions */
.modalActions {
  align-items: center;
  background: var(--palette-background-b-9);
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  box-shadow: 0px -8px 10px var(--palette-background-b-9);
  display: flex;
  margin-top: auto;
  padding: 16px 40px 40px;
  width: 100%;
  z-index: 1000;
}

.modalActionsMobile {
  padding: 8px 16px;
}
