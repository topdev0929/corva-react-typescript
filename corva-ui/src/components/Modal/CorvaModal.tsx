import classNames from 'classnames';
import { noop } from 'lodash';
import { Modal as MUIModal } from '@material-ui/core';
import { Close as CloseIcon } from '@material-ui/icons';

import { isMobileDetected, isNativeDetected } from '~/utils/mobileDetect';
import { MOBILE_MODAL_TYPES, ModalProps } from '~/components/Modal/types';
import IconButton from '~/components/IconButton';

import styles from './CorvaModal.css';

const DEFAULT_TITLES = {
  [MOBILE_MODAL_TYPES.FILTERS]: 'Filters',
  [MOBILE_MODAL_TYPES.SETTINGS]: 'Settings',
};

export function CorvaModal({
  actions,
  actionsClassName,
  children,
  contentClassName,
  contentContainerClassName,
  isCloseIconHidden = false,
  isMobile = isMobileDetected,
  isNative = isNativeDetected,
  modalTitleClassName,
  closeIconClassName,
  onClose = noop,
  open,
  rootClassName,
  size,
  stickyFooterClassName,
  title,
  type,
  'data-testid': PAGE_NAME = 'CorvaModal',
  ...otherProps
}: ModalProps): JSX.Element {
  const isMobileOrNative = isNative || isMobile;
  const modalTitle = title || DEFAULT_TITLES[type];

  return (
    <MUIModal
      open={open}
      onClose={onClose}
      className={classNames(!isMobileOrNative && 'c-modal', styles.modalWrapper, rootClassName)}
      {...otherProps}
    >
      <div
        className={classNames({
          [styles.modalContainer]: true,
          [styles.modalContainerMobile]: isMobileOrNative,
          [styles[size]]: true,
          [contentContainerClassName]: true,
        })}
      >
        {modalTitle && (
          <div
            className={classNames(
              styles.title,
              isMobileOrNative && styles.titleMobile,
              modalTitleClassName
            )}
          >
            <div className={classNames(styles.titleText)}>{modalTitle}</div>
            {(!isCloseIconHidden || type) && (
              <IconButton
                className={closeIconClassName}
                tooltipProps={{ title: 'Close' }}
                onClick={onClose}
                data-testid={`${PAGE_NAME}_closeButton`}
              >
                <CloseIcon />
              </IconButton>
            )}
          </div>
        )}

        {children && (
          <div
            className={classNames(
              styles.modalContent,
              isMobileOrNative && styles.modalContentMobile,
              contentClassName
            )}
          >
            {children}
          </div>
        )}

        {actions && (
          <div
            className={classNames(
              styles.modalActions,
              isMobileOrNative && styles.modalActionsMobile,
              stickyFooterClassName,
              actionsClassName
            )}
          >
            {actions}
          </div>
        )}
      </div>
    </MUIModal>
  );
}
