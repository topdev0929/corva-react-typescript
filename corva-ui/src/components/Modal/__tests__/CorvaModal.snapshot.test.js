import { render } from '@testing-library/react';
import { CorvaModal } from '../CorvaModal';

describe('Snapshot', () => {
  describe('Modal', () => {
    describe('open', () => {
      it('true', () => {
        const tree = render(<CorvaModal open={true} />).baseElement;
        expect(tree).toMatchSnapshot();
      });

      it('false', () => {
        const tree = render(<CorvaModal open={false} title="title" />).baseElement;
        expect(tree).toMatchSnapshot();
      });
    });

    describe('title', () => {
      it('with title', () => {
        const tree = render(<CorvaModal open={true} title="title" />).baseElement;
        expect(tree).toMatchSnapshot();
      });
    });

    describe('children', () => {
      it('as text', () => {
        const tree = render(<CorvaModal open={true} children="children" />).baseElement;
        expect(tree).toMatchSnapshot();
      });

      it('as nodes', () => {
        const tree = render(<CorvaModal open={true} children={<div>children</div>} />).baseElement;
        expect(tree).toMatchSnapshot();
      });
    });

    describe('actions', () => {
      it('as text', () => {
        const tree = render(<CorvaModal open={true} actions="actions" />).baseElement;
        expect(tree).toMatchSnapshot();
      });

      it('as nodes', () => {
        const tree = render(<CorvaModal open={true} actions={<div>actions</div>} />).baseElement;
        expect(tree).toMatchSnapshot();
      });
    });

    describe('isCloseIconHidden', () => {
      it('true', () => {
        const tree = render(<CorvaModal open={true} isCloseIconHidden={true} />).baseElement;
        expect(tree).toMatchSnapshot();
      });

      it('false', () => {
        const tree = render(<CorvaModal open={true} isCloseIconHidden={false} />).baseElement;
        expect(tree).toMatchSnapshot();
      });
    });

    describe('isNative', () => {
      it('true', () => {
        const tree = render(
          <CorvaModal open={true} isNative={true} isCloseIconHidden={true} title="title" />
        ).baseElement;
        expect(tree).toMatchSnapshot();
      });

      it('false', () => {
        const tree = render(
          <CorvaModal open={true} isNative={false} isCloseIconHidden={true} title="title" />
        ).baseElement;
        expect(tree).toMatchSnapshot();
      });
    });

    describe('isMobile', () => {
      it('true', () => {
        const tree = render(
          <CorvaModal open={true} isMobile={true} isCloseIconHidden={true} title="title" />
        ).baseElement;
        expect(tree).toMatchSnapshot();
      });

      it('false', () => {
        const tree = render(
          <CorvaModal open={true} isMobile={false} isCloseIconHidden={true} title="title" />
        ).baseElement;
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
