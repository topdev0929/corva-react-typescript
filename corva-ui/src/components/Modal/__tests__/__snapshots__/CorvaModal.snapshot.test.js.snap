// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Modal actions as nodes 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="c-modal modalWrapper"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modalContainer undefined"
      tabindex="-1"
    >
      <div
        class="modalActions"
      >
        <div>
          actions
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot Modal actions as text 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="c-modal modalWrapper"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modalContainer undefined"
      tabindex="-1"
    >
      <div
        class="modalActions"
      >
        actions
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot Modal children as nodes 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="c-modal modalWrapper"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modalContainer undefined"
      tabindex="-1"
    >
      <div
        class="modalContent"
      >
        <div>
          children
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot Modal children as text 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="c-modal modalWrapper"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modalContainer undefined"
      tabindex="-1"
    >
      <div
        class="modalContent"
      >
        children
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot Modal isCloseIconHidden false 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="c-modal modalWrapper"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modalContainer undefined"
      tabindex="-1"
    />
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot Modal isCloseIconHidden true 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="c-modal modalWrapper"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modalContainer undefined"
      tabindex="-1"
    />
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot Modal isMobile false 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="c-modal modalWrapper"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modalContainer undefined"
      tabindex="-1"
    >
      <div
        class="title"
      >
        <div
          class="titleText"
        >
          title
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot Modal isMobile true 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="modalWrapper"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modalContainer modalContainerMobile undefined"
      tabindex="-1"
    >
      <div
        class="title titleMobile"
      >
        <div
          class="titleText"
        >
          title
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot Modal isNative false 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="c-modal modalWrapper"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modalContainer undefined"
      tabindex="-1"
    >
      <div
        class="title"
      >
        <div
          class="titleText"
        >
          title
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot Modal isNative true 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="modalWrapper"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modalContainer modalContainerMobile undefined"
      tabindex="-1"
    >
      <div
        class="title titleMobile"
      >
        <div
          class="titleText"
        >
          title
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot Modal open false 1`] = `
<body
  style=""
>
  <div />
</body>
`;

exports[`Snapshot Modal open true 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="c-modal modalWrapper"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modalContainer undefined"
      tabindex="-1"
    />
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot Modal title with title 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="c-modal modalWrapper"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modalContainer undefined"
      tabindex="-1"
    >
      <div
        class="title"
      >
        <div
          class="titleText"
        >
          title
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root makeStyles-root-1"
          data-testid="IconButton"
          tabindex="0"
          title="Close"
          type="button"
          variant="default"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
